### ISSUE TITLE
GCC 10 compatibility issues

### ISSUE THREAD

**Issue by eclipseo** [sahib/rmlint#397]
Fedora Rawhide with GCC10

```
Linking Program ==> rmlint
/usr/bin/ld: librmlint.a(cfg.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(cmdline.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(formats.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(hash-utility.o):/builddir/build/BUILD/rmlint-2.9.0/lib/../lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(hasher.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(md-scheduler.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(pathtricia.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(preprocess.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(replay.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(session.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(shredder.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(traverse.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(treemerge.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(utilities.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(xattr.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(_equal.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(csv.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(fdupes.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(json.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(null.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(pretty.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(progressbar.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(py.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(sh.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(stats.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(summary.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(timestamp.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(uniques.o):/builddir/build/BUILD/rmlint-2.9.0/lib/formats/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(fts.o):/builddir/build/BUILD/rmlint-2.9.0/lib/fts/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(checksum.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(file.o):/builddir/build/BUILD/rmlint-2.9.0/lib/config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(metrohash128.o):/builddir/build/BUILD/rmlint-2.9.0/lib/checksums/../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(xxhash.o):/builddir/build/BUILD/rmlint-2.9.0/lib/checksums/xxhash/../../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(blake2b-ref.o):/builddir/build/BUILD/rmlint-2.9.0/lib/checksums/blake2/../../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(blake2bp-ref.o):/builddir/build/BUILD/rmlint-2.9.0/lib/checksums/blake2/../../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(blake2s-ref.o):/builddir/build/BUILD/rmlint-2.9.0/lib/checksums/blake2/../../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
/usr/bin/ld: librmlint.a(blake2sp-ref.o):/builddir/build/BUILD/rmlint-2.9.0/lib/checksums/blake2/../../config.h:124: multiple definition of `RM_LOG_MTX'; src/rmlint.o:/builddir/build/BUILD/rmlint-2.9.0/src/../lib/config.h:124: first defined here
```

This is related to this change in GCC 10:

> GCC now defaults to -fno-common. As a result, global variable accesses are more efficient on various targets. In C, global variables with multiple tentative definitions now result in linker errors. With -fcommon such definitions are silently merged during linking.

https://gcc.gnu.org/gcc-10/changes.html

> Default to -fno-common
> 
> A common mistake in C is omitting extern when declaring a global variable in a header file. If the header is included by several files it results in multiple definitions of the same variable. In previous GCC versions this error is ignored. GCC 10 defaults to -fno-common, which means a linker error will now be reported. To fix this, use extern in header files when declaring global variables, and ensure each global is defined in exactly one C file. As a workaround, legacy C code can be compiled with -fcommon.
> 
>   int x;  // tentative definition - avoid in header files
> 
>   extern int y;  // correct declaration in a header file

https://gcc.gnu.org/gcc-10/porting_to.html

**Comment by sahib** [comment]
Hello @eclipseo,

that should be hopefully fixed now by df26dc40.
Please  check if that fixes it for you.

**Comment by eclipseo** [comment]
Thank you, the build is indeed fixed.
